packages/perseus-editor/src/hint-editor.tsx

```diff
@@ -363,40 +363,29 @@ class CombinedHintsEditor extends React.Component<CombinedHintsEditorProps> {
         ]);
         this.props.onChange({hints: hints}, () => {
             const i = hints.length - 1;
-            // eslint-disable-next-line react/no-string-refs
-            // @ts-expect-error - TS2339 - Property 'focus' does not exist on type 'ReactInstance'.
-            this.refs["hintEditor" + i].focus();
+            this.hintEditors[i]?.focus();
         });
     };
 
     getSaveWarnings: () => any = () => {
-        return _.chain(this.props.hints)
+        return this.props.hints
             .map((hint, i) => {
-                return _.map(
-                    // eslint-disable-next-line react/no-string-refs
-                    // @ts-expect-error - TS2339 - Property 'getSaveWarnings' does not exist on type 'ReactInstance'.
-                    this.refs["hintEditor" + i].getSaveWarnings(),
-                    (issue) => "Hint " + (i + 1) + ": " + issue,
-                );
+                return this.hintEditors["hintEditor" + i]
+                    ?.getSaveWarnings()
+                    .map((issue) => "Hint " + (i + 1) + ": " + issue);
             })
-            .flatten(true)
-            .value();
+            .flat();
     };
 
-    serialize: (options?: any) => ReadonlyArray<string> = (options: any) => {
+    serialize(options?: any): ReadonlyArray<PerseusRenderer> {
         return this.props.hints.map((hint, i) => {
             return this.serializeHint(i, options);
         });
-    };
+    }
 
-    serializeHint: (index: number, options?: any) => string = (
-        index: number,
-        options: any,
-    ): string => {
-        // eslint-disable-next-line react/no-string-refs
-        // @ts-expect-error - TS2339 - Property 'serialize' does not exist on type 'ReactInstance'.
-        return this.refs["hintEditor" + index].serialize(options);
-    };
+    serializeHint(index: number, options?: any): PerseusRenderer {
```

Comment:
Refactoring `serializeHint` to be a method instead of an arrow function changes semantics - the value of `this` might be unbound.

Reply:
`serializeHint` is only called from within this component and never passed. When possible, I've been trying to convert these arrow functions on classes to standard methods to reduce some of the noise.